modules/distribution/resources/migration-1.8.0_to_1.9.0/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom18to19.java

```diff
@@ -0,0 +1,702 @@
+
+                        //Currently set to ANONYMOUS_ROLE, need to set to visible roles
+                        ServiceHolder.getRealmService().getTenantUserRealm(tenant.getId()).getAuthorizationManager()
+                                .authorizeRole(APIConstants.ANONYMOUS_ROLE,
+                                        "_system/governance" + swagger2location, ActionConstants.GET);
+                    }
+                }
+            }
+        } catch (MalformedURLException e) {
+            ResourceUtil.handleException("Error occurred while creating swagger v2.0 document ", e);
+        } catch (APIManagementException e) {
+            ResourceUtil.handleException("Error occurred while reading API from the artifact ", e);
+        } catch (RegistryException e) {
+            ResourceUtil.handleException("Error occurred while accessing the registry", e);
+        } catch (ParseException e) {
+            ResourceUtil.handleException("Error occurred while getting swagger v2.0 document", e);
+        } catch (UserStoreException e) {
+            ResourceUtil.handleException("Error occurred while reading tenant information", e);
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Swagger resource migration done for all the tenants");
+        }
+
+    }
+
+    /**
+     * This method generates swagger v2 doc using swagger 1.2 doc
+     *
+     * @param registry          governance registry
+     * @param swagger12location the location of swagger 1.2 doc
+     * @return JSON string of swagger v2 doc
+     * @throws java.net.MalformedURLException
+     * @throws org.json.simple.parser.ParseException
+     * @throws org.wso2.carbon.registry.core.exceptions.RegistryException
+     */
+
+    private String getSwagger2docUsingSwagger12RegistryResources(Registry registry, String swagger12location)
+            throws MalformedURLException, ParseException, RegistryException {
+
+        JSONParser parser = new JSONParser();
+        String swagger12BasePath = null;
+
+        Resource swaggerRes = registry.get(swagger12location + APIConstants.API_DOC_1_2_RESOURCE_NAME);
+        JSONObject swagger12doc = (JSONObject) parser.parse(new String((byte[]) swaggerRes.getContent()));
+
+        Map<String, JSONArray> apiDefPaths = new HashMap<String, JSONArray>();
+        Resource swagger12Res = registry.get(swagger12location);
+
+        //get all the resources inside the 1.2 resource location
+        String[] apiDefinitions = (String[]) swagger12Res.getContent();
+
+        //get each resource in the 1.2 folder except the api-doc resource
+        for (String apiDefinition : apiDefinitions) {
+
+            String resourceName = apiDefinition.substring(apiDefinition.lastIndexOf("/"));
+            //skip if api-doc file
+            if (resourceName.equals(APIConstants.API_DOC_1_2_RESOURCE_NAME)) {
```

Comment:
its better to have this as if (APIConstants.API_DOC_1_2_RESOURCE_NAME.equals(resourceName))